massa-bootstrap/src/tests/universe_client.rs

```diff
@@ -0,0 +1,150 @@
+            max_versioning_elements_size: MAX_BOOTSTRAP_VERSIONING_ELEMENTS_SIZE as usize,
+            max_final_state_elements_size: MAX_BOOTSTRAP_FINAL_STATE_PARTS_SIZE as usize,
+            thread_count: THREAD_COUNT,
+        }))
+            as Box<(dyn MassaDBController + 'static)>));
+        controllers
+            .final_state_controller
+            .write()
+            .expect_set_last_start_period()
+            .returning(move |_| ());
+        controllers
+            .final_state_controller
+            .write()
+            .expect_set_last_slot_before_downtime()
+            .returning(move |_| ());
+        controllers
+            .final_state_controller
+            .write()
+            .expect_get_database()
+            .return_const(database.clone());
+        let client_mip_store = MipStore::try_from_db(
+            database.clone(),
+            MipStatsConfig {
+                block_count_considered: 100,
+                warn_announced_version_ratio: Ratio::new(1, 2),
+            },
+        )
+        .unwrap();
+        controllers
+            .final_state_controller
+            .write()
+            .expect_get_mip_store_mut()
+            .returning(move || client_mip_store.clone());
+        let universe = Self {
+            controllers,
+            config,
+            global_bootstrap_state,
+            database,
+        };
+        universe.initialize();
+        universe
+    }
+}
+
+impl BootstrapClientTestUniverse {
+    pub fn launch_bootstrap(
+        &mut self,
+        remote_port: u16,
+        remote_node_id: NodeId,
+    ) -> Result<(), BootstrapError> {
+        let remote_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), remote_port);
+        self.controllers
+            .bs_connector
+            .expect_connect_timeout()
+            .times(1)
+            .returning(move |_, _| Ok(std::net::TcpStream::connect(remote_addr).unwrap()));
+        //TODO: Maybe move it out of this
+        let version = "BOOT.1.0".parse().unwrap();
+        let mut next_bootstrap_message: BootstrapClientMessage =
+            BootstrapClientMessage::AskBootstrapPart {
+                last_slot: None,
+                last_state_step: StreamingStep::Started,
+                last_versioning_step: StreamingStep::Started,
+                last_consensus_step: StreamingStep::Started,
+                send_last_start_period: true,
+            };
+
+        let mut conn = connect_to_server(
+            &mut self.controllers.bs_connector,
+            &self.config,
+            &remote_addr,
+            &remote_node_id.get_public_key(),
+            Some(self.config.rate_limit),
+        )
+        .unwrap();
+        bootstrap_from_server(
+            &self.config,
+            &mut conn,
+            &mut next_bootstrap_message,
+            &mut self.global_bootstrap_state,
+            version,
+        )
+    }
+
+    //TODO: Add consensus blocks and peers
```

Comment:
I saw that you removed the `assert_eq_bootstrap_graph`, is it better to do it differently?
If so, can you describe what you expect in an issue?

Reply:
Added in this issue : https://github.com/massalabs/massa/issues/4540